projects/lib/src/board/sittuyinboard.cpp

```diff
@@ -0,0 +1,323 @@
+{
+	return true;
+}
+
+bool SittuyinBoard::variantHasOptionalPromotions() const
+{
+	return true;
+}
+
+QList< Piece > SittuyinBoard::reservePieceTypes() const
+{
+	QList< Piece > types;
+	for (Side side: { Side::White, Side::Black} )
+	{
+		for (int type = Knight; type <= King; type++)
+			types << Piece(side, type);
+	}
+	return types;
+}
+
+MakrukBoard::CountingRules SittuyinBoard::countingRules() const
+{
+	return BareKing;
+}
+
+bool SittuyinBoard::kingsCountAssertion(int whiteKings, int blackKings) const
+{
+	Piece whiteKing(Side::White, King);
+	Piece blackKing(Side::Black, King);
+	return whiteKings + reserveCount(whiteKing) == 1 && blackKings + reserveCount(blackKing) == 1;
+}
+
+int SittuyinBoard::promotionRank(int file) const
+{
+	return std::max(file , height() - 1 - file);
+}
+
+void SittuyinBoard::addPromotions(int sourceSquare, int targetSquare, QVarLengthArray< Move >& moves) const
+{
+	moves.append(Move(sourceSquare, targetSquare, General));
+}
+
+void SittuyinBoard::generatePawnMoves(int square,
+				     QVarLengthArray<Move>& moves) const
+{
+	// Generate moves for Pawn (Feudal Lord)
+	QVarLengthArray< Move > moves1;
+	ShatranjBoard::generateMovesForPiece(moves1, Pawn, square);
+
+	// Add normal moves and demote shatranj promotions
+	for (const Move& m: moves1)
+		moves.append(Move(m.sourceSquare(), m.targetSquare()));
+
+	// No promotions if General still on board
+	Side side = sideToMove();
+	if (pieceCount(side, General) > 0)
+		return;
+
+	Square sq = chessSquare(square);
+	int file = sq.file();
+	int rank = sq.rank();
+	int rrank = (side == Side::White) ? rank : height() - 1 - rank;
+
+	// Must promote from promotion rank (except the last pawn)
+	if (rrank != promotionRank(file)
+	&&  pieceCount(sideToMove(), Pawn) != 1)
+		return;
+
+	// Generate promotion to General when on promotion square
+	addPromotions(square, square, moves);
+
+	// Generate General's moves, must not capture by promotion
+	QVarLengthArray< Move > moves2;
+	generateMovesForPiece(moves2, General, square);
+	for (const Move move: moves2)
+	{
+		if (captureType(move) == Piece::NoPiece)
+			addPromotions(square, move.targetSquare(), moves);
+	}
+}
+
+void SittuyinBoard::generateMovesForPiece(QVarLengthArray< Move >& moves,
+					  int pieceType,
+					  int square) const
+{
+	// Normal game moves
+	if (!m_inSetUp)
+	{
+		// No drops after set up
+		if (square == 0)
+			return;
+		// Pawn moves, special promotions, see above
+		if (pieceType == Pawn)
+			return generatePawnMoves(square, moves);
+
+		return MakrukBoard::generateMovesForPiece(moves, pieceType, square);
+	}
+
+	// Set-up: generate drops onto own half
+	if (square != 0)
+		return;
+
+	Side side = sideToMove();
+	const int size = arraySize();
+	const int len = (size - (width() + 2)) / 2;
+	const bool isBlack = (side == Side::Black);
+	const int start = 2 * (width() + 2) + 1;
+
+	for (int i = start; i < len; i++)
+	{
+		int index = isBlack ? i : size - 1 - i;
+		Piece tmp = pieceAt(index);
```

Comment:
1.The mirror square of index  /wrt to the board center is the square at arraySize() - 1 - index.
2. The index now starts from 21 (a8) instead of 0. 
Loop for Black: a8 ... d5 White: h1 ... e4